src/vocab/mod.rs

```diff
@@ -157,3 +159,57 @@ pub(crate) fn bracket(word: &str) -> String {
 
     bracketed
 }
+
+/// Search for the first occurrence of a value in a reverse-sorted vector.
+pub(crate) fn first_occurrence<S>(vec: &[CountedType<S>], key: &CountedType<S>) -> usize {
```

Comment:
There's also a nice crate for finding lower/upper bounds:

https://crates.io/crates/ordslice

Reply:
Apparently deprecated for:

https://crates.io/crates/superslice